### ISSUE TITLE
Build issues (module path, CUDA 9 and sm_20)

### ISSUE THREAD

**Issue by tdd11235813** [ROCmSoftwarePlatform/rocRAND#30]
- `CMAKE_MODULE_PATH=/path/to/hip/` did not work
  - tried various path combinations with `CMAKE_PREFIX_PATH` and `CMAKE_MODULE_PATH` and `HIP_DIR` as suggested by cmake output... however, only a direct `cmake -DHIP_PATH=/path/to/hip/ ..` is working
- when building with CUDA9, lowest CC with `sm_20` yields an error (CUDA 9 no more supports Fermi)
  - can be fixed in `cmake/SetupNVCC.cmake`
  - would be nice, if there would be a cmake option to propagate `NVGPU_TARGETS`
```
Building NVCC (Device) object library/CMakeFiles/hiprand.dir/src/hiprand/hiprand_generated_hiprand_nvcc.cpp.o
nvcc fatal   : Value 'sm_20' is not defined for option 'gpu-architecture'
```
after fixing this, build succeeded.
Test run successfully on P100 and GV100.

Tools: gcc/5.3.0, CUDA9.1, HIP 1.5.18315

**Comment by jszuppe** [comment]
>when building with CUDA9, lowest CC with sm_20 yields an error (CUDA 9 no more supports Fermi)

I can't reproduce this problem with CUDA 9.2 and HIP 1.5.18263. If `sm_20` is set it means that program which checks which `sm_*` should be used returned error (see https://github.com/ROCmSoftwarePlatform/rocRAND/blob/master/cmake/SetupNVCC.cmake#L25). Are you sure that `hipcc` works correctly?

> CMAKE_MODULE_PATH=/path/to/hip

I don't know what this is for. Is your HIP installation in a custom place? Try `CMAKE_MODULE_PATH=<path-to-HIP>/cmake` like it's in [here](https://github.com/ROCmSoftwarePlatform/rocRAND/blob/master/CMakeLists.txt#L31) for standard path.

**Comment by jszuppe** [comment]
You can propagate `NVGPU_TARGETS` as presented in [here](https://github.com/ROCmSoftwarePlatform/rocRAND/blob/master/cmake/SetupNVCC.cmake#L77).

**Comment by tdd11235813** [comment]
thank you for your quick response. 

The program is compiled on a login node, where no GPU is available. This explains the fallback to `sm_20`, which yields an error, when CUDA 9 is used.
Yes, setting `cmake -DNVGPU_TARGETS=...` is one way to achieve this. However, it would be nice, that this variable is cached for the cmake gui like `AMDGPU_TARGETS`, so it can be viewed and edited e.g. in `ccmake .` later on.

I forgot that the `CMAKE_MODULE_PATH` is [not an environment variable](https://gitlab.kitware.com/cmake/community/wikis/doc/cmake/Useful-Variables), thus it failed, my fault. Yes, HIP is in a custom installation path, so `CMAKE_MODULE_PATH` or `HIP_DIR` must be set.

**Comment by jszuppe** [comment]
> Yes, setting cmake -DNVGPU_TARGETS=... is one way to achieve this. However, it would be nice, that this variable is cached for the cmake gui like AMDGPU_TARGETS, so it can be viewed and edited e.g. in ccmake . later on.

You're right. I'll try to find some time to add it to make it a cached variable like `AMDGPU_TARGETS`.
